backend/server/src/db/migrations/2-id-generation.sql

```diff
@@ -0,0 +1,47 @@
+-- Set up the necessary means for our pseudo-randomized ID-generation.
+-- The basic idea is that there is a `sequence` and an XTEA encryption key
+--
+-- To use these facilities for a table, just call
+--
+-- ```sql
+-- call prepare_randomized_ids('my_entity');
+-- ```
+--
+-- where `'my_entity'` is just some descriptive name you pick.
+--
+-- Then you can use something like the following as your primary key:
+--
+-- ```sql
+-- -- [...]
+-- id primary key default randomized_id('my_entity'),
+-- -- [...]
+-- ```
+
+create extension pgcrypto;
+
+-- Create a table to hold encryption keys for the `xtea` function,
+-- one per relation where we want to "randomize" the IDs, to avoid
+-- being able to cross-guess the sequence.
+create table __keys (
```

Comment:
Minor thing: maybe `__xtea_keys` or `__sequence_encryption_keys` would be more appropriate?

Reply:
Went with the first one. :)